cli/src/merge_tools/external.rs

```diff
@@ -58,6 +58,14 @@ pub struct ExternalMergeTool {
     /// `$left`, `$right`, `$base`, and `$output` are replaced with
     /// paths to the corresponding files.
     pub merge_args: Vec<String>,
+    /// By default, if a merge tool exits with a non-zero exit code, then the
+    /// merge will be cancelled. Some merge tools allow leaving some conflicts
+    /// unresolved, in which case they will be left as conflict markers in the
+    /// output file. In that case, the merge tool may exit with a non-zero exit
+    /// code to indicate that not all conflicts were resolved. Adding an exit
+    /// code to this array will tell `jj` to interpret that exit code as
+    /// indicating that the `$output` file should contain conflict markers.
+    pub merge_tool_conflict_exit_codes: Vec<i32>,
```

Comment:
nit: the flag naming is already a bit inconsistent, but I think `merge_conflict_exit_codes` (without `tool`) is good. All parameters defined here are tool configuration.